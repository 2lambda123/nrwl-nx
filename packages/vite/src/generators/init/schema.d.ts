export interface Schema {
  uiFramework: 'react' | 'none';
}
