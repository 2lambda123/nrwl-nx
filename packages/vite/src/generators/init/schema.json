{
  "cli": "nx",
  "title": "Add Vite Configuration to the workspace",
  "description": "Add Vite Configuration to the workspace.",
  "$id": "init-vite-plugin",
  "type": "object",
  "properties": {
    "uiFramework": {
      "type": "string",
      "description": "UI Framework to use for Vite.",
      "enum": ["react", "none"],
      "default": "react",
      "x-prompt": "What UI framework plugin should Vite use?"
    }
  }
}
