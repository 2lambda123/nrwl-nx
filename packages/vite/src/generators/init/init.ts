import {
  formatFiles,
  GeneratorCallback,
  readNxJson,
  Tree,
  updateNxJson,
} from '@nx/devkit';

import { InitGeneratorSchema } from './schema';
import {
  addPlugin,
  checkDependenciesInstalled,
  moveToDevDependencies,
} from './lib/utils';

export function updateNxJsonSettings(tree: Tree) {
  const nxJson = readNxJson(tree);

  const productionFileSet = nxJson.namedInputs?.production;
  if (productionFileSet) {
    productionFileSet.push(
      '!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)',
      '!{projectRoot}/tsconfig.spec.json'
    );

    nxJson.namedInputs.production = Array.from(new Set(productionFileSet));
  }

  const hasPlugin = nxJson.plugins?.some((p) =>
    typeof p === 'string'
      ? p === '@nx/vite/plugin'
      : p.plugin === '@nx/vite/plugin'
  );

  if (!hasPlugin) {
    nxJson.targetDefaults ??= {};
    nxJson.targetDefaults['@nx/vite:test'] ??= {};
    nxJson.targetDefaults['@nx/vite:test'].cache ??= true;
    nxJson.targetDefaults['@nx/vite:test'].inputs ??= [
      'default',
      productionFileSet ? '^production' : '^default',
    ];
  }

  updateNxJson(tree, nxJson);
}

export async function initGenerator(tree: Tree, schema: InitGeneratorSchema) {
  moveToDevDependencies(tree);
  updateNxJsonSettings(tree);

  const addPlugins = process.env.NX_PCV3 === 'true';
  if (addPlugins) {
    addPlugin(tree);
  }

  let installTask: GeneratorCallback = () => {};
  if (!schema.skipPackageJson) {
    installTask = checkDependenciesInstalled(tree, schema);
  }

  if (!schema.skipFormat) {
    await formatFiles(tree);
  }

  return installTask;
}

export default initGenerator;
