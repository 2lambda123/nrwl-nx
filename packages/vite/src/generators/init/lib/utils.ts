import {
  addDependenciesToPackageJson,
  logger,
  readJson,
  readNxJson,
  Tree,
  updateJson,
  updateNxJson,
} from '@nx/devkit';

import {
  edgeRuntimeVmVersion,
  happyDomVersion,
  jsdomVersion,
  nxVersion,
  vitePluginDtsVersion,
  vitePluginReactSwcVersion,
  vitePluginReactVersion,
  vitestVersion,
  viteVersion,
} from '../../../utils/versions';
import { InitGeneratorSchema } from '../schema';

export function checkDependenciesInstalled(
  host: Tree,
  schema: InitGeneratorSchema
) {
  const packageJson = readJson(host, 'package.json');
  const devDependencies = {};
  const dependencies = {};
  packageJson.dependencies = packageJson.dependencies || {};
  packageJson.devDependencies = packageJson.devDependencies || {};

  // base deps
  devDependencies['@nx/vite'] = nxVersion;
  devDependencies['vite'] = viteVersion;
  devDependencies['vitest'] = vitestVersion;
  devDependencies['@vitest/ui'] = vitestVersion;

  if (schema.testEnvironment === 'jsdom') {
    devDependencies['jsdom'] = jsdomVersion;
  } else if (schema.testEnvironment === 'happy-dom') {
    devDependencies['happy-dom'] = happyDomVersion;
  } else if (schema.testEnvironment === 'edge-runtime') {
    devDependencies['@edge-runtime/vm'] = edgeRuntimeVmVersion;
  } else if (schema.testEnvironment !== 'node' && schema.testEnvironment) {
    logger.info(
      `A custom environment was provided: ${schema.testEnvironment}. You need to install it manually.`
    );
  }

  if (schema.uiFramework === 'react') {
    if (schema.compiler === 'swc') {
      devDependencies['@vitejs/plugin-react-swc'] = vitePluginReactSwcVersion;
    } else {
      devDependencies['@vitejs/plugin-react'] = vitePluginReactVersion;
    }
  }

  if (schema.includeLib) {
    devDependencies['vite-plugin-dts'] = vitePluginDtsVersion;
  }

  return addDependenciesToPackageJson(host, dependencies, devDependencies);
}

export function moveToDevDependencies(tree: Tree) {
  updateJson(tree, 'package.json', (packageJson) => {
    packageJson.dependencies = packageJson.dependencies || {};
    packageJson.devDependencies = packageJson.devDependencies || {};

    if (packageJson.dependencies['@nx/vite']) {
      packageJson.devDependencies['@nx/vite'] =
        packageJson.dependencies['@nx/vite'];
      delete packageJson.dependencies['@nx/vite'];
    }
    return packageJson;
  });
}

export function createVitestConfig(tree: Tree) {
  const nxJson = readNxJson(tree);

  const productionFileSet = nxJson.namedInputs?.production;
  if (productionFileSet) {
    productionFileSet.push(
      '!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)',
      '!{projectRoot}/tsconfig.spec.json'
    );

    nxJson.namedInputs.production = Array.from(new Set(productionFileSet));
  }

  nxJson.targetDefaults ??= {};
  nxJson.targetDefaults['@nx/vite:test'] ??= {};
  nxJson.targetDefaults['@nx/vite:test'].cache ??= true;
  nxJson.targetDefaults['@nx/vite:test'].inputs ??= [
    'default',
    productionFileSet ? '^production' : '^default',
  ];
  nxJson.targetDefaults['@nx/vite:test'].options ??= {
    passWithNoTests: true,
  };

  updateNxJson(tree, nxJson);
}

export function addPlugin(tree: Tree) {
  const nxJson = readNxJson(tree);
  nxJson.plugins ??= [];

  for (const plugin of nxJson.plugins) {
    if (
      typeof plugin === 'string'
        ? plugin === '@nx/vite/plugin'
        : plugin.plugin === '@nx/vite/plugin'
    ) {
      return;
    }
  }

  nxJson.plugins.push({
    plugin: '@nx/vite/plugin',
    options: {
      buildTargetName: 'build',
      previewTargetName: 'preview',
      testTargetName: 'test',
      serveTargetName: 'serve',
      serveStaticTargetName: 'serve-static',
    },
  });
  updateNxJson(tree, nxJson);
}
