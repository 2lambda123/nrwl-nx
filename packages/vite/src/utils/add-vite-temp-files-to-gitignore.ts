import type { Tree } from '@nx/devkit';

export function addViteTempFilesToGitIgnore(tree: Tree) {
  let gitIgnoreContents = `**/vite.config.{js,ts,mjs,mts,cjs,cts}.timestamp*`;
  if (tree.exists('.gitignore')) {
    gitIgnoreContents = `${tree.read('.gitignore', 'utf-8')}
    ${gitIgnoreContents}`;
  }

  tree.write('.gitignore', gitIgnoreContents);
}
