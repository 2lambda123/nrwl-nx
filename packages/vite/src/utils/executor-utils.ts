import { printDiagnostics, runTypeCheck } from '@nx/js';
import { join, resolve } from 'path';
import { ViteBuildExecutorOptions } from '../executors/build/schema';
import { ExecutorContext } from '@nx/devkit';
import { ViteDevServerExecutorOptions } from '../executors/dev-server/schema';
import {
  calculateProjectDependencies,
  createTmpTsConfig,
} from '@nx/js/src/utils/buildable-libs-utils';

export async function validateTypes(opts: {
  workspaceRoot: string;
  projectRoot: string;
  tsconfig: string;
}): Promise<void> {
  const result = await runTypeCheck({
    workspaceRoot: opts.workspaceRoot,
    tsConfigPath: join(opts.workspaceRoot, opts.tsconfig),
    mode: 'noEmit',
  });

  await printDiagnostics(result.errors, result.warnings);

  if (result.errors.length > 0) {
    throw new Error('Found type errors. See above.');
  }
}

export function createBuildableTsConfig(
  projectRoot: string,
  options: ViteBuildExecutorOptions | ViteDevServerExecutorOptions,
  context: ExecutorContext
) {
  const tsConfig = resolve(projectRoot, 'tsconfig.json');
  options.buildLibsFromSource ??= true;

  if (!options.buildLibsFromSource) {
    const { dependencies } = calculateProjectDependencies(
      context.projectGraph,
      context.root,
      context.projectName,
      context.targetName,
      context.configurationName
    );
    // this tsconfig is used via the vite ts paths plugin
    createTmpTsConfig(tsConfig, context.root, projectRoot, dependencies);
  }
}
