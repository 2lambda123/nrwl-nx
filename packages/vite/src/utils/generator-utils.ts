import {
  joinPathFragments,
  logger,
  offsetFromRoot,
  readJson,
  readProjectConfiguration,
  TargetConfiguration,
  Tree,
  updateProjectConfiguration,
  writeJson,
} from '@nrwl/devkit';
import { ViteBuildExecutorOptions } from '../executors/build/schema';
import { ViteDevServerExecutorOptions } from '../executors/dev-server/schema';
import { VitestExecutorOptions } from '../executors/test/schema';
import { Schema } from '../generators/configuration/schema';

/**
 * This function is used to find the build and serve targets for
 * an application.
 *
 * The reason this function exists is because we cannot assume
 * that the user has not created a custom build target for the application,
 * or that they have not changed the name of the build target
 * from build to anything else.
 *
 * So, in order to find the correct name of the target,
 * we have to look into all the targets, check the executor
 * they are using, and infer from the executor that the target
 * is a build target.
 */
export function findExistingTargets(targets: {
  [targetName: string]: TargetConfiguration;
}): {
  buildTarget: string;
  serveTarget: string;
  testTarget: string;
} {
  const returnObject: {
    buildTarget: string;
    serveTarget: string;
    testTarget: string;
  } = {
    buildTarget: 'build',
    serveTarget: 'serve',
    testTarget: 'test',
  };

  Object.entries(targets).forEach(([target, targetConfig]) => {
    switch (targetConfig.executor) {
      case '@nrwl/webpack:dev-server':
      case '@nxext/vite:dev':
        returnObject.serveTarget = target;
        break;
      case '@angular-devkit/build-angular:browser':
        /**
         * Not looking for '@nrwl/angular:ng-packagr-lite' or any other
         * angular executors.
         * Only looking for '@angular-devkit/build-angular:browser'
         * because the '@nrwl/angular:ng-packagr-lite' executor
         * (and maybe the other custom exeucutors) is only used for libs atm.
         */
        returnObject.buildTarget = target;
        break;
      case '@nrwl/webpack:webpack':
      case '@nrwl/next:build':
      case '@nrwl/web:webpack':
      case '@nrwl/web:rollup':
      case '@nrwl/js:tsc':
      case '@nrwl/angular:ng-packagr-lite':
      case '@nrwl/js:babel':
      case '@nrwl/js:swc':
      case '@nxext/vite:build':
        returnObject.buildTarget = target;
        break;
      case '@nrwl/jest:jest':
      case 'nxext/vitest:vitest':
        returnObject.testTarget = target;
      default:
        returnObject.buildTarget = 'build';
        returnObject.serveTarget = 'serve';
        returnObject.testTarget = 'test';
        break;
    }
  });

  return returnObject;
}

export function addOrChangeTestTarget(
  tree: Tree,
  options: Schema,
  target: string
) {
  const project = readProjectConfiguration(tree, options.project);
  const targets = {
    ...project.targets,
  };

  const testOptions: VitestExecutorOptions = {
    passWithNoTests: true,
  };

  if (targets[target]) {
    targets[target].executor = '@nrwl/vite:test';
    delete targets[target].options.jestConfig;
  } else {
    targets[target] = {
      executor: '@nrwl/vite:test',
      outputs: ['{projectRoot}/coverage'],
      options: testOptions,
    };
  }

  updateProjectConfiguration(tree, options.project, {
    ...project,
    targets: {
      ...targets,
    },
  });
}

export function addOrChangeBuildTarget(
  tree: Tree,
  options: Schema,
  target: string
) {
  const project = readProjectConfiguration(tree, options.project);

  const buildOptions: ViteBuildExecutorOptions = {
    outputPath: joinPathFragments(
      'dist',
      project.root != '.' ? project.root : options.project
    ),
  };

  const targets = {
    ...project.targets,
  };

  if (targets[target]) {
    targets[target].options = {
      ...buildOptions,
    };
    targets[target].executor = '@nrwl/vite:build';
  } else {
    targets[`${target}`] = {
      executor: '@nrwl/vite:build',
      outputs: ['{options.outputPath}'],
      defaultConfiguration: 'production',
      options: buildOptions,
      configurations: {
        development: {},
        production: {
          fileReplacements: [
            {
              replace: joinPathFragments(
                project.sourceRoot,
                'environments/environment.ts'
              ),
              with: joinPathFragments(
                project.sourceRoot,
                'environments/environment.prod.ts'
              ),
            },
          ],
          sourceMap: false,
        },
      },
    };
  }

  updateProjectConfiguration(tree, options.project, {
    ...project,
    targets: {
      ...targets,
    },
  });
}

export function addOrChangeServeTarget(
  tree: Tree,
  options: Schema,
  target: string
) {
  const project = readProjectConfiguration(tree, options.project);

  const serveOptions: ViteDevServerExecutorOptions = {
    buildTarget: `${options.project}:build`,
  };

  const targets = {
    ...project.targets,
  };

  if (targets[target]) {
    targets[target].options = {
      ...serveOptions,
    };
    targets[target].executor = '@nrwl/vite:dev-server';
  } else {
    targets[`${target}`] = {
      executor: '@nrwl/vite:dev-server',
      defaultConfiguration: 'development',
      options: {
        buildTarget: `${options.project}:build`,
      },
      configurations: {
        development: {
          buildTarget: `${options.project}:build:development`,
        },
        production: {
          buildTarget: `${options.project}:build:production`,
          hmr: false,
        },
      },
    };
  }

  updateProjectConfiguration(tree, options.project, {
    ...project,
    targets: {
      ...targets,
    },
  });
}

export function editTsConfig(tree: Tree, options: Schema) {
  const projectConfig = readProjectConfiguration(tree, options.project);

  const config = readJson(tree, `${projectConfig.root}/tsconfig.json`);

  switch (options.uiFramework) {
    case 'react':
      config.compilerOptions = {
        target: 'ESNext',
        useDefineForClassFields: true,
        lib: ['DOM', 'DOM.Iterable', 'ESNext'],
        allowJs: false,
        skipLibCheck: true,
        esModuleInterop: false,
        allowSyntheticDefaultImports: true,
        strict: true,
        forceConsistentCasingInFileNames: true,
        module: 'ESNext',
        moduleResolution: 'Node',
        resolveJsonModule: true,
        isolatedModules: true,
        noEmit: true,
        jsx: 'react-jsx',
        types: ['vite/client'],
      };
      config.include = [...config.include, 'src'];
      break;
    case 'none':
      config.compilerOptions = {
        target: 'ESNext',
        useDefineForClassFields: true,
        module: 'ESNext',
        lib: ['ESNext', 'DOM'],
        moduleResolution: 'Node',
        strict: true,
        resolveJsonModule: true,
        isolatedModules: true,
        esModuleInterop: true,
        noEmit: true,
        noUnusedLocals: true,
        noUnusedParameters: true,
        noImplicitReturns: true,
        skipLibCheck: true,
        types: ['vite/client'],
      };
      config.include = [...config.include, 'src'];
      break;
    default:
      break;
  }

  writeJson(tree, `${projectConfig.root}/tsconfig.json`, config);
}

export function moveAndEditIndexHtml(
  tree: Tree,
  options: Schema,
  buildTarget: string
) {
  const projectConfig = readProjectConfiguration(tree, options.project);

  let indexHtmlPath =
    projectConfig.targets[buildTarget].options?.index ??
    `${projectConfig.root}/src/index.html`;
  const mainPath = (
    projectConfig.targets[buildTarget].options?.main ??
    `${projectConfig.root}/src/main.ts${
      options.uiFramework === 'react' ? 'x' : ''
    }`
  ).replace(projectConfig.root, '');

  if (
    !tree.exists(indexHtmlPath) &&
    tree.exists(`${projectConfig.root}/index.html`)
  ) {
    indexHtmlPath = `${projectConfig.root}/index.html`;
  }

  if (tree.exists(indexHtmlPath)) {
    const indexHtmlContent = tree.read(indexHtmlPath, 'utf8');
    if (
      !indexHtmlContent.includes(
        `<script type="module" src="${mainPath}"></script>`
      )
    ) {
      tree.write(
        `${projectConfig.root}/index.html`,
        indexHtmlContent.replace(
          '</body>',
          `<script type="module" src="${mainPath}"></script>
          </body>`
        )
      );

      if (tree.exists(`${projectConfig.root}/src/index.html`))
        tree.delete(`${projectConfig.root}/src/index.html`);
    }
  } else {
    tree.write(
      `${projectConfig.root}/index.html`,
      `<!DOCTYPE html>
      <html lang="en">
        <head>
          <meta charset="UTF-8" />
          <link rel="icon" href="/favicon.ico" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
          <title>Vite</title>
        </head>
        <body>
          <div id="root"></div>
          <script type="module" src="${mainPath}"></script>
        </body>
      </html>`
    );
  }
}

export function writeViteConfig(tree: Tree, options: Schema) {
  const projectConfig = readProjectConfiguration(tree, options.project);

  const viteConfigPath = `${projectConfig.root}/vite.config.ts`;

  if (tree.exists(viteConfigPath)) {
    // TODO (katerina): Ideally we should check if the config is already set up correctly
    logger.info(
      `vite.config.ts already exists. Skipping creation of vite config for ${options.project}.`
    );
    return;
  }

  let viteConfigContent = '';

  const testOption = `test: {
    globals: true,
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    ${
      options.inSourceTests
        ? `includeSource: ['src/**/*.{js,mjs,cjs,ts,mts,cts,jsx,tsx}']`
        : ''
    }
  },`;

  const defineOption = `define: {
    'import.meta.vitest': undefined
  },`;

  switch (options.uiFramework) {
    case 'react':
      viteConfigContent = `
${options.includeVitest ? '/// <reference types="vitest" />' : ''}
      import { defineConfig } from 'vite';
      import react from '@vitejs/plugin-react';
      import ViteTsConfigPathsPlugin from 'vite-tsconfig-paths';
      
      export default defineConfig({
        plugins: [
          react(),
          ViteTsConfigPathsPlugin({
            root: '${offsetFromRoot(projectConfig.root)}',
            projects: ['tsconfig.base.json'],
          }),
        ],
        ${options.inSourceTests ? defineOption : ''}
        ${options.includeVitest ? testOption : ''}
      });`;
      break;
    case 'none':
      viteConfigContent = `
      ${options.includeVitest ? '/// <reference types="vitest" />' : ''}
      import { defineConfig } from 'vite';
      import ViteTsConfigPathsPlugin from 'vite-tsconfig-paths';
      
      export default defineConfig({
        plugins: [
          ViteTsConfigPathsPlugin({
            root: '${offsetFromRoot(projectConfig.root)}',
            projects: ['tsconfig.base.json'],
          }),
        ],
        ${options.inSourceTests ? defineOption : ''}
        ${options.includeVitest ? testOption : ''}
      });`;
      break;
    default:
      break;
  }

  tree.write(viteConfigPath, viteConfigContent);
}
