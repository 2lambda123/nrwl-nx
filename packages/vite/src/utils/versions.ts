export const nxVersion = require('../../package.json').version;
export const viteVersion = '^3.0.5';
export const vitePluginEslintVersion = '^1.6.0';
export const vitestVersion = '^0.19.1';
export const vitestUiVersion = '^0.9.3';
export const vitePluginReactVersion = '^2.2.0';
export const vitePluginVueVersion = '^3.2.0';
export const vitePluginVueJsxVersion = '^2.1.1';
export const viteTsConfigPathsVersion = '^3.5.2';
