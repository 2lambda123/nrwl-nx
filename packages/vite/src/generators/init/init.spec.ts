import {
  addDependenciesToPackageJson,
  NxJsonConfiguration,
  ProjectGraph,
  readJson,
  readNxJson,
  Tree,
  updateJson,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { nxVersion } from '../../utils/versions';

let projectGraph: ProjectGraph;
jest.mock('@nx/devkit', () => ({
  ...jest.requireActual<any>('@nx/devkit'),
  createProjectGraphAsync: jest.fn().mockImplementation(async () => {
    return projectGraph;
  }),
}));

import { initGenerator } from './init';

describe('@nx/vite:init', () => {
  let tree: Tree;
  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    projectGraph = {
      nodes: {},
      dependencies: {},
    };
  });

  describe('vite basic setup', () => {
    it('should add required packages', async () => {
      const existing = 'existing';
      const existingVersion = '1.0.0';
      addDependenciesToPackageJson(
        tree,
        { '@nx/vite': nxVersion, [existing]: existingVersion },
        { [existing]: existingVersion }
      );
      await initGenerator(tree, {
        addPlugin: true,
      });
      const packageJson = readJson(tree, 'package.json');
      expect(packageJson).toMatchSnapshot();
    });

    it('should add sharedGlobals for .env if sharedGlobals is defined', async () => {
      updateJson<NxJsonConfiguration>(tree, 'nx.json', (json) => {
        json.namedInputs ??= {};
        json.namedInputs.sharedGlobals = [];
        return json;
      });
      await initGenerator(tree, {});
      const sharedGlobalsNamedInputs = readJson(tree, 'nx.json').namedInputs
        .sharedGlobals;
      console.log(sharedGlobalsNamedInputs);
      expect(sharedGlobalsNamedInputs).toContain('{workspaceRoot}/.env?(.*)');
    });
  });

  describe('vitest targets', () => {
    it('should add target defaults for test', async () => {
      updateJson<NxJsonConfiguration>(tree, 'nx.json', (json) => {
        json.namedInputs ??= {};
        json.namedInputs.production = ['default'];
        return json;
      });

      await initGenerator(tree, {
        addPlugin: true,
      });

      const nxJson = readNxJson(tree);

      expect(nxJson).toMatchInlineSnapshot(`
        {
          "affected": {
            "defaultBase": "main",
          },
          "namedInputs": {
            "production": [
              "default",
              "!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)",
              "!{projectRoot}/tsconfig.spec.json",
            ],
          },
          "plugins": [
            {
              "options": {
                "buildTargetName": "build",
                "previewTargetName": "preview",
                "serveStaticTargetName": "serve-static",
                "serveTargetName": "serve",
                "testTargetName": "test",
              },
              "plugin": "@nx/vite/plugin",
            },
          ],
          "targetDefaults": {
            "build": {
              "cache": true,
            },
            "lint": {
              "cache": true,
            },
          },
        }
      `);
    });
  });
});
