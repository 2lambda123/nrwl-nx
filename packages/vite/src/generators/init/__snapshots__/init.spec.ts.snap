// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nx/vite:init dependencies for package.json should add vite packages and react-related dependencies for vite 1`] = `
{
  "dependencies": {
    "existing": "1.0.0",
  },
  "devDependencies": {
    "@nx/vite": "0.0.1",
    "@vitejs/plugin-react": "^4.2.0",
    "@vitest/ui": "^1.0.4",
    "existing": "1.0.0",
    "vite": "^5.0.0",
    "vitest": "^1.0.4",
  },
  "name": "@proj/source",
}
`;

exports[`@nx/vite:init dependencies for package.json should support --testEnvironment=edge-runtime 1`] = `
{
  "dependencies": {},
  "devDependencies": {
    "@edge-runtime/vm": "~3.0.2",
    "@nx/vite": "0.0.1",
    "@vitest/ui": "^1.0.4",
    "vite": "^5.0.0",
    "vitest": "^1.0.4",
  },
  "name": "@proj/source",
}
`;

exports[`@nx/vite:init dependencies for package.json should support --testEnvironment=happy-dom 1`] = `
{
  "dependencies": {},
  "devDependencies": {
    "@nx/vite": "0.0.1",
    "@vitest/ui": "^1.0.4",
    "happy-dom": "~9.20.3",
    "vite": "^5.0.0",
    "vitest": "^1.0.4",
  },
  "name": "@proj/source",
}
`;

exports[`@nx/vite:init dependencies for package.json should support --testEnvironment=jsdom 1`] = `
{
  "dependencies": {},
  "devDependencies": {
    "@nx/vite": "0.0.1",
    "@vitest/ui": "^1.0.4",
    "jsdom": "~22.1.0",
    "vite": "^5.0.0",
    "vitest": "^1.0.4",
  },
  "name": "@proj/source",
}
`;
